eva-lib/src/main/java/uk/ac/ebi/eva/lib/repository/VariantEntityRepositoryImpl.java

```diff
@@ -111,12 +124,102 @@ public Long countByRegionsAndComplexFilters(List<Region> regions, List<VariantEn
             exclude.forEach(e -> query.fields().exclude(e));
         }
 
-        return mongoTemplate.find(query, VariantEntity.class);
+        List<VariantEntity> variantEntities = mongoTemplate.find(query, VariantEntity.class);
+
+        Table<String, String, Map<String, String>> studyFileIdsToPositionSamples = studyFileIdsToPositionSamples();
+
+        variantEntities = updateVariantEntitiesSampleNames(variantEntities, studyFileIdsToPositionSamples);
+
+        return variantEntities;
+    }
+
+    private Table<String, String, Map<String, String>> studyFileIdsToPositionSamples() {
+        List<VariantSourceEntity> variantSourceEntities = variantSourceEntityRepository.findAll();
 
+        Table<String, String, Map<String, String>> studyFileIdsToPositionSamples = HashBasedTable.create();
+
+        for (VariantSourceEntity variantSourceEntity : variantSourceEntities) {
+            if (variantSourceEntity.getSamplesPosition() == null) {
+                continue;
+            }
+
+            String studyId = variantSourceEntity.getStudyId();
+            String fileId = variantSourceEntity.getFileId();
+
+            Map<String, Integer> samplesPosition = variantSourceEntity.getSamplesPosition();
+            Map<String, String> positionSamples = new HashMap<>();
+            for(Map.Entry<String, Integer> entry : samplesPosition.entrySet()) {
+                positionSamples.put(Integer.toString(entry.getValue()), entry.getKey());
```

Comment:
samplesPosition won't contain "def", samplesPosition is the data from the files collection. If you look at the declaration for samplesPosition `Map<String, Integer> samplesPosition`, the value is guaranteed to be an Integer.